src/Optimizations/peps.jl

```diff
@@ -0,0 +1,96 @@
+using AutoHOOT, ChainRulesCore, Zygote
+using ..ITensorAutoHOOT
+using ..ITensorNetworks
+using ITensors: setinds
+using ..ITensorNetworks: PEPS, randomizePEPS!, inner_network, extract_data
+using ..ITensorAutoHOOT: batch_tensor_contraction
+
+# TODO: rewrite this function
+function ChainRulesCore.rrule(::typeof(ITensors.prime), peps::PEPS; ham=true)
```

Comment:
Agreed this should be rewriiten.

It seems like if the generic `mapinds` function was used: https://github.com/ITensor/ITensorNetworkAD.jl/blob/c178a13c63f95f8bc51e322f16eb6ab217b7217b/src/ITensorNetworks/tensor_networks.jl#L641-L657

then is could just be defined so that the pullback is written in terms of `mapinds(x -> prime(x, -1), linkinds, dpeps_prime.data)`.

Reply:
Done